shared/src/main/java/org/jetbrains/skija/Surface.java

```diff
@@ -4,37 +4,388 @@
+     * <p>Copies each readable pixel intersecting both rectangles, without scaling,
+     * converting to Surface colorType() and Surface alphaType() if required.</p>
+     *
+     * @param bitmap storage for pixels to copy to Surface
+     * @param x      x-axis position relative to Surface to begin copy; may be negative
+     * @param y      y-axis position relative to Surface to begin copy; may be negative
+     * @see <a href="https://fiddle.skia.org/c/@Surface_writePixels_2">https://fiddle.skia.org/c/@Surface_writePixels_2</a>
+     */
+    public void writePixels(Bitmap bitmap, int x, int y) {
+        Stats.onNativeCall();
+        _nWritePixels(_ptr, Native.getPtr(bitmap), x, y);
+    }
+
+    /**
+     * <p>Call to ensure all reads/writes of the surface have been issued to the underlying 3D API.</p>
+     *
+     * <p>Skia will correctly order its own draws and pixel operations.
+     * This must to be used to ensure correct ordering when the surface backing store is accessed
+     * outside Skia (e.g. direct use of the 3D API or a windowing system).
+     * DirectContext has additional flush and submit methods that apply to all surfaces and images created from
+     * a DirectContext.
+     */
+    public void flushAndSubmit() {
+        Stats.onNativeCall();
+        _nFlushAndSubmit(_ptr, false);
+    }
+
+    /**
+     * <p>Call to ensure all reads/writes of the surface have been issued to the underlying 3D API.</p>
+     *
+     * <p>Skia will correctly order its own draws and pixel operations.
+     * This must to be used to ensure correct ordering when the surface backing store is accessed
+     * outside Skia (e.g. direct use of the 3D API or a windowing system).
+     * DirectContext has additional flush and submit methods that apply to all surfaces and images created from
+     * a DirectContext.
+     *
+     * @param syncCpu a flag determining if cpu should be synced
+     */
+    public void flushAndSubmit(boolean syncCpu) {
+        Stats.onNativeCall();
+        _nFlushAndSubmit(_ptr, syncCpu);
+    }
+
+    public void flush() {
+        Stats.onNativeCall();
+        _nFlush(_ptr);
+    }
+
+    /**
+     * <p>May return true if the caller is the only owner.</p>
+     *
+     * <p>Ensures that all previous owner's actions are complete.</p>
+     */
+    public boolean isUnique() {
+        Stats.onNativeCall();
+        return _nUnique(_ptr);
+    }
+
+    /**
+     * <p>Increment the reference count.</p>
+     *
+     * <p>Must be balanced by a call to unref().</p>
+     */
+    public void ref() {
+        Stats.onNativeCall();
+        _nRef(_ptr);
+    }
+
+    /**
+     * <p>Decrement the reference count.</p>
+     *
+     * <p>If the reference count is 1 before the decrement, then delete the object.
+     * Note that if this is the case, then the object needs to have been allocated via new, and not on the stack.</p>
+     */
+    public void unref() {
+        Stats.onNativeCall();
+        _nUnref(_ptr);
+    }
```

Comment:
Same argument about ref/unref being only needed in C++, not in Java

Reply:
Done